### PR TITLE
fix: incorrect logging

### PR DESCRIPTION
**Pull request by MichaelMure**

### CODE REVIEW

**Review by aschmahmann — APPROVED** [review]

File: core/corehttp/gateway_handler.go [hunk 1]
```diff
@@ -854,7 +854,7 @@ func webError(w http.ResponseWriter, message string, err error, defaultCode int)
 func webErrorWithCode(w http.ResponseWriter, message string, err error, code int) {
 	http.Error(w, fmt.Sprintf("%s: %s", message, err), code)
 	if code >= 500 {
-		log.Warnf("server error: %s: %s", err)
+		log.Warnf("server error: %s: %s", message, err)
```
Comment (aschmahmann) [inline core/corehttp/gateway_handler.go:857]: The linter is complaining about this being unsanitized. IIUC these are all escaped URLs, right @lidel?

Comment (aschmahmann) [reply core/corehttp/gateway_handler.go:857]: @lidel, @guseggert, @MichaelMure if any of you get a chance to confirm the linter is off here please comment and anyone with merge access can come by and stamp it.

Thanks @MichaelMure for catching this 🙏.

Comment (MichaelMure) [reply core/corehttp/gateway_handler.go:857]: The linter is right:
- https://github.com/ipfs/go-ipfs/blob/cbb962852f914dc682887ea86c227e8f546c880c/core/corehttp/gateway_handler.go#L301
- https://github.com/ipfs/go-ipfs/blob/cbb962852f914dc682887ea86c227e8f546c880c/core/corehttp/gateway_handler.go#L200

Comment (aschmahmann) [reply core/corehttp/gateway_handler.go:857]: Does the path escaping from `EscapedPath` not help us out here, and we should be doing our own escaping?

Comment (MichaelMure) [reply core/corehttp/gateway_handler.go:857]: I can't say for sure, but this escaping looks more about encoding for a URL path rather than security.

Comment (lidel) [reply core/corehttp/gateway_handler.go:857]: - klinda false-positive: `webError` is returned as `text/plain` so it does not need to be protected against HTML injection (we don't have to wrap it in HTMLEscapeString)

**Review by lidel — COMMENTED** [review]
Thanks! 
To avoid doing the same work twice, I fixed this as part of bigger refactor in  https://github.com/ipfs/go-ipfs/pull/8758 / https://github.com/ipfs/go-ipfs/pull/8758/commits/eb95d2b693ad3ccf028d3995b9b109307dae9f62 so for sake of keeping smaller review surface let's close this and focus on https://github.com/ipfs/go-ipfs/pull/8758

**Comment by lidel** [comment]
Continued in https://github.com/ipfs/go-ipfs/pull/8758

**Pull request closed without merging**